hdate/date.py

```diff
@@ -205,6 +208,82 @@ def omer_day(self):
             return 0
         return omer_day
 
+    @property
+    def next_day(self):
+        """Returns the HDate for the next day."""
+        return HDate(self.gdate + datetime.timedelta(+1), self.diaspora,
+                     self.hebrew)
+
+    @property
+    def previous_day(self):
+        """Returns the HDate for the previous day."""
+        return HDate(self.gdate + datetime.timedelta(-1), self.diaspora,
+                     self.hebrew)
+
+    def get_upcoming_shabbat_hdate(self):
+        """Returns the HDate for either the upcoming or current Shabbat.
+
+        If it is currently Shabbat, returns the HDate of the Saturday.
+        """
+        today = self.gdate
+        if today.weekday() == 6:
+            # If it's Sunday, fast forward to the next Shabbat.
```

Comment:
Can be simplified: saturday = self.gdate + timedelta((12 - self.gdate.weekday()) % 7)

Reply:
That takes care of Sunday as well